client/src/Features/MicrosoftLearn/MicrosoftLearnFilterUtils.ts

```diff
@@ -0,0 +1,145 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Product, Role, Level, Catalog } from '../../Models/Learn';
+import { FilterType } from '../../Models/Learn/FilterType.model';
+import { LearnTypeFilterOption, FilterOption } from './MicrosoftLearnFilterComponentProps';
+import _ from 'lodash';
+import { getParentProduct, scoreRegex } from './MicrosoftLearnFilterCore';
+import { Filter } from '../../Models/Learn/Filter.model';
+
+const FilterOptionComparer = (a: FilterOption, b: FilterOption) => {
+  if (a && b) {
+    return a.name.localeCompare(b.name);
+  } else if (a) {
+    return 1;
+  }
+  return -1;
+};
+
+const createOptionArrayFromKeys = (keys: FilterOption[]) => {
```

Comment:
The return type is `Map<FilterOption, FilterOption[]>`. Essentially, since `Roles`, `Types`, `Levels` are currently non-hierarchical, to parse the available array of the corresponding `FilterOption`s to a common map format, I am using this function. In retrospect, `createOptionsMapFromKeys` would be a more suitable name.

Reply:
There's a `toMap` utility created in our codebase, does it seems like a usecase for that?

Reply:
Yes, in fact, that's exactly what we need here. Made the necessary changes!